cocos/ui/UIPageView.cpp

```diff
@@ -145,12 +145,24 @@ void PageView::setCurPageIndex( ssize_t index )
 
 void PageView::setCurrentPageIndex(ssize_t index)
 {
+	auto preindex = MAX(0, getCurrentPageIndex());
+		
     jumpToItem(index, Vec2::ANCHOR_MIDDLE, Vec2::ANCHOR_MIDDLE);
+
+	if (preindex != index)
+	{
+		pageTurningEvent();
+	}
 }
 
 void PageView::scrollToPage(ssize_t idx)
 {
-    scrollToItem(idx);
+	auto preindex = MAX(0, getCurrentPageIndex());
```

Comment:
After calling scrollToItem, it ListView will scroll to the specific item, when the scroll ended , it will fire the `Page turning` event, so there is no need to fire the event manually.